### PR TITLE
Code Block

### PR DESCRIPTION
**Pull request by Blackbaud-SteveBrush**

### CODE REVIEW

File: src/modules/code-block/code-block.component.ts [hunk 1]
```diff
@@ -14,26 +15,74 @@ export class StacheCodeBlockComponent implements AfterViewInit {
...
+  public get languageType(): string {
+    return this._languageType;
+  };
 
   @ViewChild('codeFromContent')
   public codeTemplateRef: any;
 
   public output: string;
+  private readonly defaultLanguage: string = 'markup';
+  private readonly validLanguages: string[] = [
```
Comment (blackbaud-brandonhare) [inline src/modules/code-block/code-block.component.ts]: We are using prism for this right? Do they have some sort of list of supported languages that we can import or check against? Should we put this in a service? Having the languages hardcoded feels weird.

File: src/modules/code-block/code-block.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  it('should not honor angular bindings in the inner HTML', () => {
+    const code = '<p>Hey! {{name}}</p>';
+    const testFixture = TestBed.createComponent(StacheCodeBlockTestComponent);
+    const testElement = testFixture.nativeElement;
+    testFixture.detectChanges();
+    expect(testElement.querySelector('.stache-code-output').innerText).toContain(code);
+  });
+
+  it('should handle invalid language types', () => {
+    const code = '<p>/p>';
```
Comment (blackbaud-brandonhare) [inline src/modules/code-block/code-block.component.spec.ts]: Missing an opening bracket on the closing p tag.

**Comment by Blackbaud-SteveBrush** [comment]
Ready for another round, @blackbaud-brandonhare

File: src/modules/code-block/code-block.component.ts [hunk 3]
```diff
@@ -14,26 +15,63 @@ export class StacheCodeBlockComponent implements AfterViewInit {
...
   public codeTemplateRef: any;
 
   public output: string;
+  private readonly defaultLanguage: string = 'markup';
+  private validLanguages: string[];
+  private _languageType: string = this.defaultLanguage;
+
+  public constructor(
+    private cdRef: ChangeDetectorRef) {
+      this.validLanguages = Object.keys(Prism.languages);
```
Comment (blackbaud-brandonhare) [inline src/modules/code-block/code-block.component.ts]: MUCH Better!

**Pull request merged**